scripts/check_ecr_findings.sh

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env bash
+
+# Exit on any error
+set -eu -o pipefail
+
+# Set the repository name and image tag
+repository_name="$1"
+image_tag="$2"
+
+# Wait for the scan to complete
+echo "Waiting for scan to complete..."
+while true; do
+    # Temporarily disable exit on error
+    set +e
+    # Capture both stdout and stderr, redirecting stderr to stdout
+    scanFindings=$(aws ecr describe-image-scan-findings --repository-name "$repository_name" --image-id imageTag="$image_tag" 2>&1)
+    scanExitCode=$?
+    # Re-enable exit on error
+    set -e
+
+    # Check for RepositoryNotFoundException
+    if echo "$scanFindings" | grep -q "RepositoryNotFoundException"; then
+        echo "Error: Repository not found."
+        exit 1
+    fi
+
+    if [[ "$scanExitCode" -eq 254 ]]; then
+        echo "Scan not available yet, retrying..."
+        sleep 10
+        continue # Use continue to retry only for code 254
+    elif [[ "$scanExitCode" -ne 0 ]]; then
+        echo "An error occurred with exit code $scanExitCode."
+        echo "Error details: $scanFindings"
+        exit $scanExitCode
+    fi
+
+    # If we get here, it means exit code was 0, and we can check the scan status
+    scanStatus=$(echo "$scanFindings" | jq -r '.imageScanStatus.status')
+
+    if [[ "$scanStatus" == "COMPLETE" ]]; then
+        echo "Scan complete"
+        break
+    elif [[ "$scanStatus" == "IN_PROGRESS" ]]; then
+        echo "Scan in progress..."
+        sleep 10
+    elif [[ "$scanStatus" == "FAILED" ]]; then
+        echo "Scan failed"
+        exit 1
+    else
+        echo "Unexpected scan status: $scanStatus"
+        exit 1
+    fi
+done
+
+# Retrieve the scan findings and parse the total findings
+totalFindings=$(aws ecr describe-image-scan-findings --repository-name "$repository_name" --image-id imageTag="$image_tag" | jq '.imageScanFindings.findingSeverityCounts | add')
+
+# Check if totalFindings is not null and greater than 0
+if [[ "$totalFindings" != "null" && "$totalFindings" -gt 0 ]]; then
+  echo "Scan found $totalFindings findings!"
+  exit 1
+else
+  echo "Scan found no findings"
+fi
```

Comment:
We should probably move this into the `while` block, since we already have a variable in that scope for the scan's findings (`$scanFindings`) and shouldn't need to re-run it here.